src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int MAX_RANGE = 20;
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomNameFigure = new Random().nextInt(FiguresName.values().length);
+        switch (FiguresName.values()[randomNameFigure]) {
+            case CIRCLE:
+                Figure circle = new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANGE) + 1);
+                return circle;
+            case SQUARE:
+                Figure square = new Square(colorSupplier.getRandomColor(),
```

Comment:
`colorSupplier.getRandomColor()` can be moved to a variable before the `switch`

Reply:
not fixed